index.js

```diff
@@ -11,16 +11,18 @@ const isForced = "FORCE_COLOR" in env || argv.includes("--color")
 const isWindows = platform === "win32"
 const isDumbTerminal = env.TERM === "dumb"
 
+const isTty = tty && tty.isatty && tty.isatty(1)
+
 const isCompatibleTerminal =
-  tty && tty.isatty && tty.isatty(1) && env.TERM && !isDumbTerminal
+  isTty && env.TERM && !isDumbTerminal
 
 const isCI =
   "CI" in env &&
   ("GITHUB_ACTIONS" in env || "GITLAB_CI" in env || "CIRCLECI" in env)
 
 export const isColorSupported =
   !isDisabled &&
-  (isForced || (isWindows && !isDumbTerminal) || isCompatibleTerminal || isCI)
+  (isForced || (isWindows && !isDumbTerminal && isTty) || isCompatibleTerminal || isCI)
```

Comment:
Could we achieve the same result by checking isCompatibleTerminal earlier?

Reply:
I think not, `isCompatibleTerminal` will return `false` both when run through non-TTY GUI app and when through command line directly on Windows. And it seems like Windows console is never TTY either while checked through nodejs.